site/src/lib/api/blocks/npm.ts

```diff
@@ -0,0 +1,278 @@
+    cwd: packageFolder,
+    nocase: true,
+  })[0];
+  if (!metadataJsonPath) {
+    throw new Error("No block-metadata.json present in package", {
+      cause: { code: "INVALID_PACKAGE_CONTENTS" },
+    });
+  }
+
+  let metadataJson;
+  try {
+    const metadataJsonString = fs
+      .readFileSync(path.resolve(packageFolder, metadataJsonPath))
+      .toString();
+    metadataJson = JSON.parse(metadataJsonString);
+  } catch (err) {
+    throw new Error(
+      `Could not parse block-metadata.json: ${(err as Error).message}`,
+      {
+        cause: { code: "INVALID_PACKAGE_CONTENTS" },
+      },
+    );
+  }
+
+  const blockSourceFolder = path.resolve(
+    packageFolder,
+    path.dirname(metadataJsonPath),
+  );
+
+  // move the readme into the block's source folder, if it exists
+  const readmePath = glob.sync("README.md", {
+    cwd: packageFolder,
+    nocase: true,
+  })[0];
+  if (readmePath) {
+    fs.renameSync(
+      path.resolve(packageFolder, readmePath),
+      path.resolve(blockSourceFolder, path.basename(readmePath)),
+    );
+  }
+
+  // check if we have 'example-graph.json', which we then construct a URL for in the extended matadata
+  const includesExampleGraph = await fs.pathExists(
+    path.resolve(blockSourceFolder, "example-graph.json"),
+  );
+
+  // check block-metadata.json contains required properties
+  for (const key of ["blockType", "protocol", "schema", "source", "version"]) {
+    if (!metadataJson[key]) {
+      throw new Error(`block-metadata.json must contain a '${key}' property`, {
+        cause: { code: "INVALID_PACKAGE_CONTENTS" },
+      });
+    }
+  }
+
+  // check that the source file actually exists
+  const sourcePath = metadataJson.source;
+  const sourceFileExists = await fs.pathExists(
+    path.resolve(blockSourceFolder, sourcePath),
+  );
+
+  if (!sourceFileExists) {
+    throw new Error(
+      `block-metadata.json 'source' path '${sourcePath}' does not exist`,
+      {
+        cause: { code: "INVALID_PACKAGE_CONTENTS" },
+      },
+    );
+  }
+
+  // check that the schema actually exists
+  const schemaPath = metadataJson.schema;
+  if (schemaPath.startsWith("http")) {
+    await fetch(schemaPath, { method: "HEAD" });
+  } else {
+    const schemaFileExists = await fs.pathExists(
+      path.resolve(blockSourceFolder, schemaPath),
+    );
+    if (!schemaFileExists) {
+      throw new Error(
+        `block-metadata.json 'schema' path '${schemaPath}' does not exist`,
+        {
+          cause: { code: "INVALID_PACKAGE_CONTENTS" },
+        },
+      );
+    }
+  }
+
+  const now = new Date().toISOString();
+
+  // for blocks developed locally, add a prefix to the storage URL - the R2 bucket is shared across all dev environments
+  let storageNamespacePrefix = "";
+  if (!isRunningOnVercel) {
+    const exec = promisify(child_process.exec);
+
+    const gitConfigUserNameResult = await exec("git config --get user.name");
```

Comment:
We might be able to use `execa` here, especially given that it’s already important. If we cannot, a comment explaining why would be useful.

Reply:
done in https://github.com/blockprotocol/blockprotocol/pull/522/commits/db9a55ba883429bc77f70d061d2b78d51aba1e80